community/learn/graphql-tutorials/tutorials/typescript-react-apollo/tutorial-site/content/queries/2-create-query.md

```diff
@@ -0,0 +1,176 @@
+---
+title: "useQuery Hook"
+metaTitle: "Apollo Query with useQuery Hook | GraphQL React Apollo Typescript Tutorial"
+metaDescription: "We will use the Apollo useQuery Hook from @apollo/react-hooks. This hook is used to fetch data and return data, loading and error"
+---
+
+import GithubLink from "../../src/GithubLink.js";
+
+In this section, we will implement GraphQL Queries and integrate with the react UI.
+With Apollo Client, you can send queries in 3 different ways.
+
+1. Using the `client.query` method directly and then process the response.
+2. Render Prop API with React.
+3. Using the `useQuery` hook. (Recommended)
+
+### Apollo useQuery Hook
+The recommended method is to use the `useQuery` hook, where you will just wrap your GraphQL query to the `gql` function and will fetch the data automatically and will give a result object containing data, loading and error properties.
+
+Great! Now let's define the graphql query to be used:
+
+Open `src/components/Todo/TodoPrivateList.tsx` and add the following code:
+
+<GithubLink link="https://github.com/hasura/graphql-engine/blob/master/community/learn/graphql-tutorials/tutorials/typescript-react-apollo/app-final/src/components/Todo/TodoPrivateList.tsx" text="src/components/Todo/TodoPrivateList.tsx" />
+
+```javascript
+  import React, { Fragment, useState } from "react";
++ import gql from 'graphql-tag';
+  import TodoItem from "./TodoItem";
+  import TodoFilters from "./TodoFilters";
+
++ const GET_MY_TODOS = gql`
++  query getMyTodos {
++    todos(where: { is_public: { _eq: false} }, order_by: { created_at: desc }) {
++      id
++      title
++      is_completed
++  }
++ }`;
+
+  type Todo = {
+    id: number,
+    title: string,
+    is_completed: boolean
+  };
+
+```
+
+We have now written the graphql query as a javascript constant using the `gql` parser function. This function is used to parse the plain string as a graphql query.
+
+What does this query do? 
+------------------------
+The query fetches `todos` with a simple condition; `is_public` must be false. We sort the todos descending by its `created_at` time according to the schema. We specify which fields we need for the todos node.
```

Comment:
This*

Also, consider moving the `What does this query do` section to the previous section where we made them try out the query?